flang/lib/Optimizer/CodeGen/CodeGen.cpp

```diff
@@ -2891,24 +2865,17 @@ struct UnboxCharOpConversion : public FIROpConversion<fir::UnboxCharOp> {
   }
 };
 
-// unbox a procedure box value, yielding its components
+/// Lower `fir.unboxproc` operation. Unbox a procedure box value, yielding its
+/// components.
+/// TODO: Part of supporting Fortran 2003 procedure pointers.
 struct UnboxProcOpConversion : public FIROpConversion<fir::UnboxProcOp> {
   using FIROpConversion::FIROpConversion;
 
   mlir::LogicalResult
-  matchAndRewrite(fir::UnboxProcOp unboxproc, OperandTy operands,
+  matchAndRewrite(fir::UnboxProcOp unboxproc, OpAdaptor adaptor,
                   mlir::ConversionPatternRewriter &rewriter) const override {
-    auto *ctx = unboxproc.getContext();
-    auto loc = unboxproc.getLoc();
-    auto tuple = operands[0];
-    auto ty = tuple.getType();
-    mlir::Value ptr =
-        genExtractValueWithIndex(loc, tuple, ty, rewriter, ctx, 0);
-    mlir::Value host =
-        genExtractValueWithIndex(loc, tuple, ty, rewriter, ctx, 1);
-    llvm::SmallVector<mlir::Value> repls{ptr, host};
-    rewriter.replaceOp(unboxproc, repls);
-    return success();
+    TODO(unboxproc.getLoc(), "fir.unboxproc codegen");
+    return failure();
```

Comment:
Diverge. Again this looks plausible, could be useful documentation...

Reply:
Done.